service/src/main/java/greencity/service/ubs/UBSManagementServiceImpl.java

```diff
@@ -157,24 +157,27 @@ public void returnOverpayment(Long orderId,
             .orElseThrow(
                 () -> new NotFoundException(ORDER_WITH_CURRENT_ID_DOES_NOT_EXIST + orderId));
         checkAvailableOrderForEmployee(order, email);
+        if (!OrderStatus.DONE.equals(order.getOrderStatus()) && !OrderStatus.CANCELED.equals(order.getOrderStatus())) {
+            throw new BadRequestException(BAD_ORDER_STATUS_REQUEST + order.getOrderStatus());
+        }
         Payment payment = createPayment(order, overpaymentInfoRequestDto);
-        if ((order.getOrderStatus() == OrderStatus.DONE)) {
+        if (OrderStatus.DONE.equals(order.getOrderStatus())) {
             returnOverpaymentForStatusDone(user, order, overpaymentInfoRequestDto, payment);
         }
-        if (order.getOrderStatus() == OrderStatus.CANCELED
-            && overpaymentInfoRequestDto.getComment().equals(AppConstant.PAYMENT_REFUND)) {
-            returnOverpaymentAsMoneyForStatusCancelled(user, order, overpaymentInfoRequestDto);
-            collectEventsAboutOverpayment(overpaymentInfoRequestDto.getComment(), order, email);
-        }
-        if (order.getOrderStatus() == OrderStatus.CANCELED && overpaymentInfoRequestDto.getComment()
-            .equals(AppConstant.ENROLLMENT_TO_THE_BONUS_ACCOUNT)) {
-            returnOverpaymentAsBonusesForStatusCancelled(user, order, overpaymentInfoRequestDto);
-            collectEventsAboutOverpayment(overpaymentInfoRequestDto.getComment(), order, email);
-        }
-        if (OrderStatus.CANCELED.equals(order.getOrderStatus())
-            && !AppConstant.PAYMENT_REFUND.equals(overpaymentInfoRequestDto.getComment())
-            && !AppConstant.ENROLLMENT_TO_THE_BONUS_ACCOUNT.equals(overpaymentInfoRequestDto.getComment())) {
-            throw new NotFoundException(COMMENT_ERROR + overpaymentInfoRequestDto.getComment());
+        if (OrderStatus.CANCELED.equals(order.getOrderStatus())) {
+            if (overpaymentInfoRequestDto.getComment().equals(AppConstant.PAYMENT_REFUND)) {
+                returnOverpaymentAsMoneyForStatusCancelled(user, order, overpaymentInfoRequestDto);
+                collectEventsAboutOverpayment(overpaymentInfoRequestDto.getComment(), order, email);
+            }
+            if (overpaymentInfoRequestDto.getComment()
+                .equals(AppConstant.ENROLLMENT_TO_THE_BONUS_ACCOUNT)) {
+                returnOverpaymentAsBonusesForStatusCancelled(user, order, overpaymentInfoRequestDto);
+                collectEventsAboutOverpayment(overpaymentInfoRequestDto.getComment(), order, email);
+            }
+            if (!AppConstant.PAYMENT_REFUND.equals(overpaymentInfoRequestDto.getComment())
+                && !AppConstant.ENROLLMENT_TO_THE_BONUS_ACCOUNT.equals(overpaymentInfoRequestDto.getComment())) {
+                throw new NotFoundException(COMMENT_ERROR + overpaymentInfoRequestDto.getComment());
+            }
```

Comment:
Could we make here switch case construction? Orders status always is fixed, so I think that better to use here switch case construction instead of if else